Fix (not) showing values for empty cells in wxDVC using wxDataViewVirtualListModel in wxGTK

vadz opened the pull request:
This should fix #22359 and, AFAICS, doesn't have any negative effects.

I couldn't find any real explanation about why was this check added in the first place, but there doesn't seem to be any performance reason to not make it even for the virtual list models: it is called for every item in the model, which makes it totally unsuitable for big numbers of items, but this was already the case before anyhow and the extra overhead of calling `HasValue()` for it shouldn't be noticeable compared to calling `GetValue()`. But please let me know if I'm missing anything here or, also, if there are any other situations in which this still doesn't fix the original problem.

vadz committed:
Don't show value-less wxDataViewVirtualListModel cells in wxGTK

For some reason, calls to wxGtkTreeSetVisibleProp() were skipped when
using virtual list model in wxGTK implementation, resulting in showing
the value of the previous (i.e. upper) cell for the rows of this model
for which no value was available.

Simply remove IsVirtualListModel() checks and always set the cell
visibility to fix this.

This commit is best viewed ignoring whitespace-only changes.

vadz committed:
Inline wxGtkTreeSetVisibleProp() function

No real changes, just get rid of a trivial helper function which is only
used once since the changes of the previous commit and copy its code
directly into the caller.

vslavik reviewed (approved):
Thanks! This looks good to me w.r.t. correctness.

Could it be further simplified by removing the `HasValue()` check and `if` statement, as `PrepareForItem()` already does that via`CheckedGetValue()`? I'm not saying it *should* be done, but checking if my understanding is correct and this condition is just an optimization.

I *think* that as it is now (and this logic is shared by some other ports), there's a very subtle, mostly academical difference between `!HasValue()` and not returning a value from `GetValue()`:

1. If `GetValue()` doesn't return a value, `PrepareForItem()` [still enables/disables cell state]( https://github.com/wxWidgets/wxWidgets/blob/8245ba96ee5ce3e19564eabc6976435ab9ae906f/src/common/datavcmn.cpp#L914) even for empty cells - apparently intentionally.
2. But if `HasValue()` returns false, `PrepareForItem()` is skipped altogether and enabled state is not adjusted.

vadz committed:
Remove duplicated HasValue() call from wxGTK wxDataViewCtrl code

HasValue() is already called by PrepareForItem(), so there is no need to
call it explicitly from wxGTK code, just rely on PrepareForItem()
returning false if there is no value to show -- we can skip the call to
GtkSetCurrentItem() in this case, this function is cheap, and we lose
more by calling HasValue() twice in the common case than we save on not
calling it.

No real changes.

vadz commented:
Yes, I think you're right, thanks, it looks better to simplify this even if I don't think this can make any difference in practice because `SetEnabled()` is only used by the custom renderers AFAICS and their `Render()` shouldn't be called anyhow if the cell is invisible. Done now.

BTW, concerning the "apparently intentionally" part, I think 361c6357b4 (Use wxDataViewRenderer::PrepareForItem() in all ports, 2015-08-29) just prevented the existing behaviour (but added an explicit comment about it), so I'm not totally certain if this is needed neither, but I'm definitely not going to change it without a good reason -- and this call is cheap, so it shouldn't result in any performance problems neither.
